app/extensions/safe/network/authenticator-comms.js

```diff
@@ -76,6 +79,42 @@ export const handleSafeAuthAuthentication = ( uriOrReqObject, type ) =>
 
 export const getPeruseAuthReqUri = () => browserAuthReqUri;
 
+export const onNetworkStateChange = (store) => (state) =>
+{
+  logger.info('net callback state: ', state);
+  const previousState = store.getState().peruseApp.networkStatus;
+  logger.info('previousState: ', previousState);
+  store.dispatch( setNetworkStatus( state ) );
+  // TODO/BOOKMARK: Expected is an updated store state but not happening
+  logger.info('store state: ', store.getState());
+  const newState = store.getState().peruseApp.networkStatus;
```

Comment:
@hunterlester the `dispatch` calls happen async, so here, when you're pinging for `getState` again, it hasn't necessarily updated the store.

But you don't need to ping the store for this anyway, the passed `state` to this function is your newState.